frontend/src/utils/blockUtils.js

```diff
@@ -26,11 +26,13 @@ export const genJSON = (block, id) => {
   };
 };
 
-export function generateId(block) {
-  const nanoid = customAlphabet("1234567890abcedfghijklmnopqrstuvwxyz", 12);
+export function generateId(prefix) {
+  const nanoid = customAlphabet("abcdefghijklmnopqrstuvwxyz0123456789", 12);
+
   const newNanoid = nanoid();
-  const id = `${block.information.id}-${newNanoid}`;
-  return id;
+  const basePrefix = prefix ? prefix.split("-")[0] : "pipeline";
```

Comment:
Block ids look like: `canny-edge`, so I don't think we should be splitting them. Pipelines look like `pipeline-12321312aglk`. So you may want to split to a variable and then check if it's `pipeline`, if it is, use that prefix, if it's not, use the non-split version.

Reply:
Changed:

```
export function generateId(prefix) {
  const nanoid = customAlphabet("abcdefghijklmnopqrstuvwxyz0123456789", 12);

  const newNanoid = nanoid();

  if (prefix) {
    const [basePrefix] = prefix.split("-");
    const finalPrefix = basePrefix === "pipeline" ? basePrefix : prefix;
    return `${finalPrefix}-${newNanoid}`;
  }

  return `pipeline-${newNanoid}`;
}
```